controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -778,10 +796,12 @@ protected void executeSingleTargetChild(PathAddress base, PathElement currentEle
 
         @Override
         protected boolean authorize(OperationContext context, PathAddress base, ModelNode operation) {
-//            if (base.size() > 0) {
-                //We should always be able to read the root resource
+            try {
                 context.readResource(base, false);
-//            }
+            } catch(IllegalArgumentException ex) {
```

Comment:
This is kind of overkill but I wonder if it would be better if ControllerLogger.unknownRole() threw some sort of custom RuntimeException instead of IAE. Documented in the RoleMapper.canRunAs javadoc.

Catching general purpose exceptions that aren't documented smells funny. Granted, OperationContext.readResource wouldn't document the exception either, so really the only benefit of a custom exception is to avoid catching unrelated problems.

WDYT?